library/core/src/mem/mod.rs

```diff
@@ -1288,11 +1288,15 @@ impl<T> SizedTypeProperties for T {}
 
 /// Expands to the offset in bytes of a field from the beginning of the given type.
 ///
-/// Only structs, unions and tuples are supported.
+/// Structs, enums, unions and tuples are supported.
 ///
 /// Nested field accesses may be used, but not array indexes like in `C`'s `offsetof`.
 ///
-/// Note that the output of this macro is not stable, except for `#[repr(C)]` types.
+/// Enum variants may be traversed as if they were fields. Variants themselves do
+/// not have an offset.
+///
+/// Note that type layout is, in general, [platform-specific, and subject to
```

Comment:
Nit: I would emphasize that type layout is unstable by putting "subject to change" or similar wording first and then mentioning the platform-specific nature second. It might also be useful to call out `#[repr(C)]` but I don't feel strongly about that.